indexers/src/services/hyperbridge.service.ts

```diff
@@ -0,0 +1,232 @@
+import { HYPERBRIDGE_STATS_ENTITY_ID } from "../constants";
+import { SupportedChain } from "../types";
+import {
+  PostRequestEventLog,
+  PostResponseEventLog,
+} from "../types/abi-interfaces/EthereumHostAbi";
+import { HyperBridgeStats, Relayer, Transfer } from "../types/models";
+import { HyperBridgeChainStatsService } from "./hyperbridgeChainStats.service";
+import assert from "assert";
+import { isHexString } from "ethers/lib/utils";
+import { EthereumHostAbi__factory } from "../types/contracts";
+
+export class HyperBridgeService {
+  /**
+   * Get the HyperBridgeStats entity
+   */
+  static async getStats(): Promise<HyperBridgeStats> {
+    let stats = await HyperBridgeStats.get(HYPERBRIDGE_STATS_ENTITY_ID);
+
+    if (!stats) {
+      stats = HyperBridgeStats.create({
+        id: HYPERBRIDGE_STATS_ENTITY_ID,
+        numberOfMessagesSent: BigInt(0),
+        numberOfSuccessfulMessagesSent: BigInt(0),
+        numberOfFailedMessagesSent: BigInt(0),
+        numberOfTimedOutMessages: BigInt(0),
+        numberOfUniqueRelayers: BigInt(0),
+        feesPayedOutToRelayers: BigInt(0),
+        protocolFeesEarned: BigInt(0),
+        totalTransfersIn: BigInt(0),
+      });
+
+      await stats.save();
+    }
+
+    return stats;
+  }
+
+  /**
+   * Perform the necessary actions related to Hyperbridge stats when a PostRequest/PostResponse event is indexed
+   */
+  static async handlePostRequestOrResponseEvent(
+    chain: SupportedChain,
+    event: PostRequestEventLog | PostResponseEventLog,
+  ): Promise<void> {
+    assert(
+      event.args,
+      "No handlePostRequestEvent/handlePostResponseEvent args",
+    );
+
+    const { args, address, transaction } = event;
+    const { status } = await transaction.receipt();
+    let { data } = args;
+
+    const protocolFee = await this.computeProtocolFeeFromHexData(address, data);
+
+    Promise.all([await this.incrementProtocolFeesEarned(protocolFee, chain)]);
+
+    if (status === false) {
```

Comment:
These metrics should not be collected here. we collect them in the transaction handlers.

Reply:
Done